db/compaction_job.cc

```diff
@@ -1260,10 +1260,11 @@ Status CompactionJob::OpenCompactionOutputFile(
 #endif  // !ROCKSDB_LITE
   // Make the output file
   unique_ptr<WritableFile> writable_file;
-  bool syncpoint_arg = env_options_.use_direct_writes;
+  EnvOptions opt_env_opts =
+      env_->OptimizeForCompactionTableWrite(env_options_, db_options_);
```

Comment:
No need to call `OptimizeForCompactionTableWrite()`. It is now called on db open and `SetOptions()` if `bytes_per_sync` changed. You are welcome to add comment here to mention it.